File: LeetCode/Java Solutions/Fraction Addition and Subtraction/fraction.java
class Solution {
    public String fractionAddition(String expression) {
        // Add '+' at the beginning if the expression doesn't start with a sign
        if (expression.charAt(0) != '+' && expression.charAt(0) != '-') {
            expression = "+" + expression;
        }
        
        // Parse fractions
        List<int[]> fractions = new ArrayList<>();
        int i = 0;
        while (i < expression.length()) {
            int sign = expression.charAt(i) == '+' ? 1 : -1;
            i++;
            int j = i;
            while (j < expression.length() && expression.charAt(j) != '+' && expression.charAt(j) != '-') {
                j++;
            }
            String[] parts = expression.substring(i, j).split("/");
            int num = Integer.parseInt(parts[0]);
            int den = Integer.parseInt(parts[1]);
            fractions.add(new int[]{sign * num, den});
            i = j;
        }
        
        // Sum all fractions
        int numerator = 0, denominator = 1;
        for (int[] fraction : fractions) {
            int num = fraction[0], den = fraction[1];
            numerator = numerator * den + num * denominator;
            denominator *= den;
            int g = gcd(Math.abs(numerator), denominator);
            numerator /= g;
            denominator /= g;
        }
        
        // Convert result to string format
        return numerator + "/" + denominator;
    }
    
    private int gcd(int a, int b) {
        return b == 0 ? a : gcd(b, a % b);
    }
}

File: LeetCode/Python Solutions/Fraction Addition and Subtraction/fraction.py
from math import gcd
from fractions import Fraction

class Solution:
    def fractionAddition(self, expression: str) -> str:
        # Add '+' at the beginning if the expression doesn't start with a sign
        if expression[0] not in '+-':
            expression = '+' + expression
        
        # Parse fractions
        fractions = []
        i = 0
        while i < len(expression):
            sign = 1 if expression[i] == '+' else -1
            i += 1
            j = i
            while j < len(expression) and expression[j] != '+' and expression[j] != '-':
                j += 1
            num, den = map(int, expression[i:j].split('/'))
            fractions.append(Fraction(sign * num, den))
            i = j
        
        # Sum all fractions
        result = sum(fractions)
        
        # Convert result to string format
        return f"{result.numerator}/{result.denominator}"